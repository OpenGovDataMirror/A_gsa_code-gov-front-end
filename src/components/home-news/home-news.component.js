import React, { Component } from 'react'

export default class HomeNews extends Component {
  render() {
    return (
      <section className="tablet:grid-col-6">
        <div className="bg-primary-lighter">
          <header className="padding-x-4  padding-top-6">
            <h2 className="font-heading-l text-primary-darker text-center">
              Open Source News & Events
            </h2>
          </header>
          <ul className="padding-x-1 padding-bottom-2">
            {this.props.newsItems &&
              this.props.newsItems.map(news => (
                <li className="bg-white grid-row margin-1 padding-2" key={news.title}>
                  <div className="flex-1">
                    <h3 className="font-heading-sm text-bold">
                      <a href={news.url} target="_blank" rel="noopener noreferrer">
                        {news.title}
                      </a>
                    </h3>
                    <p className="margin-bottom-0 font-body-2xs margin-top-105">
                      {news.description}
                    </p>
                    {/*
                    <time className="font-body-3xs">{news.date}</time>
                    */}
                  </div>
                  <picture className="order-first padding-right-2">
                    <source srcSet={news.image} media="min-width: 800px" />
                    <img src={news.image} alt={news.alt} className="maxw-15 margin-right-1" />
                  </picture>
                </li>
              ))}
          </ul>
        </div>
      </section>
    )
  }
}
