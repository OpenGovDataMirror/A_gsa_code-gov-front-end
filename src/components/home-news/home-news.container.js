import { connect } from 'react-redux'
import { getConfigValue } from 'utils/other'
import HomeNews from './home-news.component'

export const mapStateToProps = () => ({
  newsItems: getConfigValue('content.home.news')
})

export default connect(mapStateToProps)(HomeNews)
