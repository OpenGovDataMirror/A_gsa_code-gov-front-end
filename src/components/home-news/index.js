import HomeNewsContainer from './home-news.container'

export default HomeNewsContainer
