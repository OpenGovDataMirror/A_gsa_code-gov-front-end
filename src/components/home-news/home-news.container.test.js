import { mapStateToProps } from 'components/home-news/home-news.container'

describe('containers - HomeNews', () => {
  describe('mapStateToProps', () => {
    it('should return the correct properties', () => {
      expect(mapStateToProps()).toMatchSnapshot()
    })
  })
})
