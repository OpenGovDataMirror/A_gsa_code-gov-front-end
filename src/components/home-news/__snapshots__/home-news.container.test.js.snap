// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers - HomeNews mapStateToProps should return the correct properties 1`] = `
Object {
  "newsItems": Array [
    Object {
      "alt": "Federal Source Code Summit",
      "date": "10/08/2020",
      "description": "Code.gov will host its first annual Federal Source Code Summit...",
      "image": "/assets/img/news/summit.jpeg",
      "title": "Federal Source Code Summit: Towards a Source Code Community",
      "url": "https://digital.gov/event/2020/10/08/federal-source-code-summit-building-coding/",
    },
    Object {
      "alt": "Frontend Updates Improve Accessibility",
      "date": "08/27/2020",
      "description": "Last month we celebrated the 30th anniversary of the Americans with Disabilities Act...",
      "image": "/assets/img/news/accessibility.jpeg",
      "title": "Front-End Updates Provide Greater Accessibility for America’s Code",
      "url": "https://medium.com/codedotgov/all-access-code-gov-website-front-end-updates-provide-greater-accessibility-for-americas-code-e00fc5e76163",
    },
    Object {
      "alt": "Know an Engineer",
      "date": "08/13/2020",
      "description": "We’re back with another installment of Better Know an Engineer...",
      "image": "/assets/img/news/rayvn.jpeg",
      "title": "Better Know an Engineer: Rayvn Manuel",
      "url": "https://medium.com/codedotgov/code-gov-better-know-an-engineer-rayvn-manuel-46188c95218b",
    },
  ],
}
`;
